ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/DiscountController.java

```diff
@@ -0,0 +1,138 @@
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/discount")
+@RestController
+@Slf4j
+public class DiscountController {
+	@Autowired
+	private DiscountService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Discount
+	 * @param Discount Data to insert
+	 * @return New Discount create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<DiscountDTO> createDiscount(@Valid @RequestBody DiscountDTO DiscountRequest) {
+		log.info("Discount to create: " + DiscountRequest);
+		ResponseEntity<DiscountDTO> response = null;
+		Discount newDiscount = this.service.create(DiscountMapper.mapper.toEntity(DiscountRequest));
+
+		if(newDiscount != null) {
+			response = ResponseEntity.ok(DiscountMapper.mapper.toDTO(newDiscount));
+			log.info("Discount created: " + newDiscount);
+		} else {
+			response = ResponseEntity.badRequest().build();
+			log.info("Error to create Discount: " + DiscountRequest);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements the HTTP Get to find all Discounts
+	 * @return List of Discounts
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@GetMapping
+	public ResponseEntity<List<DiscountDTO>> getAllDiscounts() {
+		log.info("Get All Discounts ...");
+		ResponseEntity<List<DiscountDTO>> response = null;
+		List<Discount> listDiscounts = this.service.getAll();
+
+		if(listDiscounts != null) {
+			log.info("Discounts[" + listDiscounts.size() + "]");
+			response = ResponseEntity.ok(listDiscounts.stream().map(DiscountMapper.mapper::toDTO).toList());
+		} else {
+			response = ResponseEntity.notFound().build();
+			log.info("Error to get all Discounts");
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements HTTP Get to find a Discount by id
+	 * @param id Discount Id
+	 * @return Discount data
+	 * @return New Discount create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@GetMapping("/{id}")
+	public ResponseEntity<DiscountDTO> getDiscount(@PathVariable Integer id) {
+		log.info("Get Discount[" + id + "]");
+		return service.get(id).map((Discount Discount) -> {
+			log.info("Discount found: " + Discount);
+			return ResponseEntity.ok(DiscountMapper.mapper.toDTO(Discount));
+		}).orElseGet(() -> {
+			log.warn("Discount not found: " + id);
+			return ResponseEntity.notFound().build();
+		});
+	}
+
+	/**
+	 * Implements HTTP Put for update a Discount
+	 * @param id Discount Id
+	 * @param Discount Discount data to update
+	 * @return Discount modified
+	 * @return New Discount create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@PutMapping("/{id}")
+	public ResponseEntity<DiscountDTO> updateDiscount(@PathVariable Integer id, @Valid @RequestBody DiscountDTO Discount) {
+		log.info("Discount[" + id + "] to update: " + Discount);
+		ResponseEntity<DiscountDTO> response = null;
+		Discount updateDiscountData = this.service.update(id, DiscountMapper.mapper.toEntity(Discount));
+
+		if(updateDiscountData != null) {
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.